File: benchmarks/faastroute/original/sentiment-analysis/sentiment-sns/func.py
from rpc import RPC
import base64
import requests
import json
import random
import os

def function_handler(context):
    if context["request_type"] == "GRPC":
        event = json.loads(context["request"])
        #response = requests.get(url = 'http://' + OF_Gateway_IP + ':' + OF_Gateway_Port + '/function/sha>
        #    "Subject": 'Negative Review Received',
        #    "Message": 'Review (ID = %i) of %s (ID = %i) received with negative results from sentiment a>
        #    event['reviewType'], int(event['productID']), int(event['customerID']), event['feedback'])
        #})

        response = RPC(os.environ["SENTIMENT_DB"], [json.dumps({
            'sentiment': event['sentiment'],
            'reviewType': event['reviewType'],
            'reviewID': event['reviewID'],
            'customerID': event['customerID'],
            'productID': event['productID'],
            'feedback': event['feedback']
        })], context["workflow_id"])
        return response, 200
    else:
        print("Empty request", flush=True)
        return "{}", 200

File: benchmarks/faastroute/original/serverless-election/election-gateway/func.js
const require('./rpc')

const function_handler = async (context) => {
	if (context["request_type"] != "GRPC") {
		var body = context["request"];
		if (body['requestType'] ==  'get_results') {
			return rpc.RPC(process.env.ELECTION_GET_RESULTS, [body], context["workflow_id"])[0], 200;
		}
		else if (body['requestType'] == 'vote') {
			return rpc.RPC(process.env.ELECTION_VOTE_ENQUEUER, [body], context["workflow_id"])[0], 200;
		}
		return 'invalid request type', 200;
	}
}

// Export the function
module.exports = { function_handler };
